### PR TITLE
fixes Progress Bar Visibility change with Orientation Change and login process terminated (#4001)

### LINKED ISSUE #4001: Progress Bar Visibility change with Orientation Change and login process terminated
**Summary:**

In Login activity when I clicked login button Progress Bar visible when I rotated the screen Progress bar invisible 
and login process of user stop or terminated  

**What did you expect the app to do, and what did you see instead?**
Maintaining Progress Bar Visibility  and prevent stop of login process with Orientation Change

**Steps to reproduce:**
 Go to Login activity
 Filled the information
 Click on the login button and change the rotated the screen

**Device and Android version:** 
Device :Redmi Y3;

**Commons app version:** 
 2.13.2
**Screen-shots:** 

**Would you like to work on the issue?**
**YES**
Please let us know whether you want to fix the issue by yourself. If not, anyone can get the issue assigned to them.
**YES I want to fix this**

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**
Fixes #4001 
What changes did you make and why?
**Bug** is due to the restart of LoginActivity on the configuration change 
* Overrides the onSaveInstanceState method and onRestoreInstanceState method in LoginActivity.java
* In onSaveInstanceState save  ProgressDailog state or visiblity and text of TextView(errorMessage)
* In onRestoreInstanceState restore All saved Data

**Tests performed (required)**
Tested prodDebug on Redmi Y3 level 29

**Screenshots (for UI changes only)**

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
Anyone review my pull request #4028

File: app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java [hunk 1]
```diff
@@ -456,4 +459,48 @@ public static void startYourself(Context context) {
...
+        return  passwordEdit.getText().toString();
+  }
+
+    @Override
+    protected void onRestoreInstanceState(final Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+
+        usernameEdit.setText(savedInstanceState.getString(Save_username));
+        passwordEdit.setText(savedInstanceState.getString(Save_password));
+
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java:491]: Could you look into the formatting please? There are random whitespaces where they should not exist.

File: app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java [hunk 2]
```diff
@@ -456,4 +459,48 @@ public static void startYourself(Context context) {
...
+        if(savedInstanceState.getBoolean(Save_progressDailog)){
+            performLogin();
+        }
+
+        String errorMessage=savedInstanceState.getString(Save_errorMessage);
+
+        if(sessionManager.isUserLoggedIn()){
+            showMessage(R.string.login_success, R.color.primaryDarkColor);
+        }else{
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java:501]: Please refer to codestyle, there should be whitespaces here and in other `else` statements.

**Review by misaochan — CHANGES_REQUESTED** [review]
Formatting needs a bit of work

**Comment by Prince-kushwaha** [comment]
Ok i change the codestyle according to docs

**Review by ashishkumar468 — COMMENTED** [review]
Looks good to me @Prince-kushwaha, I see a few irrelevant white spaces please fix them

**Comment by Prince-kushwaha** [comment]
ok I fixes this

**Pull request merged**